src/rust/receiver-mock/src/main.rs

```diff
@@ -68,41 +63,55 @@ pub async fn main() {
         },
     };
 
-    let stats = Statistics {
-        metrics: 0,
-        logs: 0,
-        logs_bytes: 0,
-        p_metrics: 0,
-        p_logs: 0,
-        p_logs_bytes: 0,
-        ts: time::get_now(),
-        metrics_list: HashMap::new(),
-        metrics_ip_list: HashMap::new(),
-        logs_ip_list: HashMap::new(),
-        url: format!("http://{}:{}/receiver", hostname, port),
-    };
-    let statistics = Arc::new(Mutex::new(stats));
-
-    run_app(statistics, port, opts).await;
+    run_app(hostname, port, opts).await
 }
 
-async fn run_app(stats: Arc<Mutex<Statistics>>, port: u16, opts: Options) {
-    let addr = SocketAddr::from(([0, 0, 0, 0], port));
-    println!("Receiver mock is waiting for enemy on 0.0.0.0:{}!", port);
-    let make_svc = make_service_fn(|conn: &AddrStream| {
-        let statistics = stats.clone();
-        let address = conn.remote_addr().ip();
+async fn run_app(hostname: String, port: u16, opts: Options) -> std::io::Result<()> {
+    let app_state = web::Data::new(router::AppState {
+        metrics: Mutex::new(0),
+        logs: Mutex::new(0),
+        logs_bytes: Mutex::new(0),
+        metrics_list: Mutex::new(HashMap::new()),
+        metrics_ip_list: Mutex::new(HashMap::new()),
+        logs_ip_list: Mutex::new(HashMap::new()),
+    });
 
-        async move {
-            let result =
-                service_fn(move |req| router::handle(req, address, statistics.clone(), opts));
-            Ok::<_, Infallible>(result)
-        }
+    let t = timer::Timer::new();
+    // TODO: configure interval?
```

Comment:
We could make it configurable

Reply:
Ok but firstly I'd do it in a separate PR (hence the `TODO`) and secondly I guess we'd like to have the stats printed "with configured interval but scaled to 'per minute'" (as it's done currently). What do you think @sumo-drosiek ?

Reply:
yes, we need DPM for the metrics and per second stats for logs

Reply:
added a todo in `router::start_print_stats_timer` for this

Reply:
imho we should convert todos to github issues